server.go

```diff
@@ -39,6 +42,17 @@ func (s *Server) Stop() {
 }
 
 func NewServer(runner *TaskRunner, config *Config) (*Server, error) {
+	//check if socket file exists
+	if _, err := os.Stat(config.SocketPath); os.IsExist(err) {
```

Comment:
In the scenario you are testing for, `os.Stat` will return without error because the socket file exists, and thus `os.IsExist` will not catch. You could probably do something instead like this:
```go
if _, err := os.Stat(config.SocketPath); err == nil {...}
```